substrate/client/transaction-pool/tests/zombienet/mod.rs

```diff
@@ -0,0 +1,151 @@
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! The zombienet spawner for integration tests for a transaction pool. Holds shared logic used
+//! across integration tests for transaction pool.
+
+use anyhow::anyhow;
+use tokio::sync::OnceCell;
+use txtesttool::scenario::ScenarioBuilder;
+use zombienet_sdk::{
+	subxt::{OnlineClient, SubstrateConfig},
+	LocalFileSystem, Network, NetworkConfig, NetworkConfigExt,
+};
+
+pub const ASSET_HUB_LOW_POOL_LIMIT_FATP_SPEC_PATH: &'static str =
+	"tests/zombienet/network-specs/asset-hub-low-pool-limit-fatp.toml";
+pub const ASSET_HUB_HIGH_POOL_LIMIT_FATP_SPEC_PATH: &'static str =
+	"tests/zombienet/network-specs/asset-hub-high-pool-limit-fatp.toml";
+pub const ASSET_HUB_HIGH_POOL_LIMIT_OLDP_3_COLLATORS_SPEC_PATH: &'static str =
+	"tests/zombienet/network-specs/asset-hub-high-pool-limit-oldp-3-collators.toml";
+pub const ASSET_HUB_HIGH_POOL_LIMIT_OLDP_4_COLLATORS_SPEC_PATH: &'static str =
+	"tests/zombienet/network-specs/asset-hub-high-pool-limit-oldp-4-collators.toml";
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+	#[error("Network initialization failure: {0}")]
+	NetworkInit(anyhow::Error),
+	#[error("Node couldn't be found as part of the network: {0}")]
+	NodeNotFound(anyhow::Error),
+	#[error("Failed to get node online client")]
+	FailedToGetOnlineClinet,
+	#[error("Failed to get node blocks stream")]
+	FailedToGetBlocksStream,
+}
+
+static LOGGER: OnceCell<()> = OnceCell::const_new();
+async fn init_logger() {
+	LOGGER
+		.get_or_init(|| async {
+			let _ = env_logger::try_init_from_env(
+				env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+			);
+		})
+		.await;
+}
+
+/// Result of work related to network spawning.
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Provides logic to spawn a network based on a Zombienet toml file.
+pub struct NetworkSpawner {
+	network: Network<LocalFileSystem>,
+}
+
+impl NetworkSpawner {
+	/// Initialize the network spawner based on a Zombienet toml file
+	pub async fn from_toml_with_env_logger(toml_path: &'static str) -> Result<NetworkSpawner> {
+		init_logger().await;
+		let net_config = NetworkConfig::load_from_toml(toml_path).map_err(Error::NetworkInit)?;
+		Ok(NetworkSpawner {
+			network: net_config
+				.spawn_native()
+				.await
+				.map_err(|err| Error::NetworkInit(anyhow!(err.to_string())))?,
+		})
+	}
+
+	/// Returns the spawned network.
+	pub fn network(&self) -> &Network<LocalFileSystem> {
+		&self.network
+	}
+
+	/// Returns a node client and waits for blocks productio to kick-off.
+	pub async fn wait_collator_client(
```

Comment:
_collator_ is redundant here, we can wait for any node. 
I would change the name to indicate that we are waiting for node to start producing blocks, maybe:
`wait_for_block_production` ?